File: cpp/program42.cpp
// program78.c

#include<iostream>
using namespace std;

class Array
{
    public:
        int iSize; 
        int *Arr;

    Array(int X)
    {
        iSize = X;
        Arr = new int[iSize];
    }

    void Accept()
    {
        cout<<"Enter elements of Array"<<"\n";
        int iCnt = 0;

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            cin>>Arr[iCnt];
        }
    }

    void OddDisplay()
    {
        cout<<"Odd elements are :"<<"\n";
        int iCnt = 0;

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            if(((((((((((((((((((Arr[iCnt] % 2) != 0))))))))))))))))))
            {
                cout<<Arr[iCnt]<<"\n";
            }
        }
    }
    ~Array()
    {
        delete[] Arr;
    }
};


int main()
{
    int iValue =  0;

    cout<<"Enter number of elements"<<"\n";
    cin>>iValue;

    Array aobj(iValue);
    aobj.Accept();
    aobj.OddDisplay();

    return 0;
}

File: java/program42.java
import java.util.*;

class ArrayX
{
    protected int Arr[];

    public ArrayX(int iSize)
    {
        Arr = new int[iSize];
    }

    protected void Accept()
    {
        System.out.println("Enter elements of the array");

        Scanner sobj = new Scanner(System.in);

        for(int icnt = 0; icnt < Arr.length; icnt++)
        {
            Arr[icnt] = sobj.nextInt();
        }
    }

    protected void Display()
    {
        System.out.println("Elements of the array are :");

        for(int icnt = 0; icnt < Arr.length; icnt++)
        {
            System.out.print(Arr[icnt]+"\t");
        }
        System.out.println();
    }
}

class MyArray extends ArrayX
{
    public MyArray(int iSize)
    {
        super(iSize);
    }

    public void DisplayOdd()
    {
        System.out.println("Odd elements of the Array are : ");
        
        for(int icnt = 0; icnt < Arr.length; icnt++)
        {
            if((Arr[icnt]%2) != 0)
            {
                System.out.println(Arr[icnt]);
            }
        }
    }
}

class program42
{
    public static void main(String af[])
    {
        int iValue = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of elements");
        iValue = sobj.nextInt();

        MyArray obj = new MyArray(iValue);
        obj.Accept();
        obj.Display();
        obj.DisplayOdd();
    }
}